applications/StructuralMechanicsApplication/python_scripts/kratos_main_structural.py

```diff
@@ -51,12 +52,12 @@
     gid_output.ExecuteInitialize()
 
 ## Creation of the Kratos model (build sub_model_parts or submeshes)
-StructureModel = {ProjectParameters["problem_data"]["model_part_name"].GetString(): main_model_part}
-
-## Get the list of the sub_model_parts in where the processes are to be applied
+StructureModel = Model()
+StructureModel.AddModelPart(main_model_part)
+# Get the list of the sub_model_parts in where the processes are to be applied
 for i in range(ProjectParameters["solver_settings"]["processes_sub_model_part_list"].size()):
```

Comment:
you don't need to do this. Once you add the StructureModel, it adds alone all of its submodelparts. That's actually the point of it... (one of)

Reply:
The problem is to keep to retrocompatibilty with what is done, this is the only way

Reply:
I know the submodelparts are added automatically, but if you want to do it that way we need to change all processes, that can be done in the future, but is a lot of work